src/main/java/teammates/ui/controller/AdminSessionsPageAction.java

```diff
@@ -30,141 +29,116 @@
-        String startMin = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTMINUTE);
-        String endMin = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_ENDMINUTE);
-        String timeZone = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_TIMEZONE);
+    private ActionResult validateParametersAndCreateShowPageResultIfInvalid() {
+        ZoneId zone = Const.DEFAULT_TIME_ZONE;
+        LocalDateTime start = LocalDateTime.now(zone).minusDays(3);
+        LocalDateTime end = LocalDateTime.now(zone).plusDays(4);
 
-        Date start;
-        Date end;
-        double zone = 0.0;
+        if (checkAllParameters("null")) {
+            this.rangeStart = start;
+            this.rangeEnd = end;
+            this.timeZone = zone;
+            return null;
+        }
 
-        Calendar calStart = TimeHelper.now(zone);
-        Calendar calEnd = TimeHelper.now(zone);
-        calStart.add(Calendar.DAY_OF_YEAR, -3);
-        calEnd.add(Calendar.DAY_OF_YEAR, 4);
+        if (checkAllParameters("notNull")) {
+            String startDate = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTDATE);
+            String endDate = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_ENDDATE);
+            String startHour = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTHOUR);
+            String endHour = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_ENDHOUR);
+            String startMin = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTMINUTE);
+            String endMin = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_ENDMINUTE);
+            String timeZone = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_TIMEZONE);
 
-        if (checkAllParameters("null")) {
-            start = calStart.getTime();
-            end = calEnd.getTime();
-        } else if (checkAllParameters("notNull")) {
-
-            SanitizationHelper.sanitizeForHtml(startDate);
-            SanitizationHelper.sanitizeForHtml(endDate);
-            SanitizationHelper.sanitizeForHtml(startHour);
-            SanitizationHelper.sanitizeForHtml(endHour);
-            SanitizationHelper.sanitizeForHtml(startMin);
-            SanitizationHelper.sanitizeForHtml(endMin);
-            SanitizationHelper.sanitizeForHtml(timeZone);
-
-            zone = Double.parseDouble(timeZone);
-
-            start = TimeHelper.convertLocalDateTimeToDate(
-                    TimeHelper.parseLocalDateTime(startDate, startHour, startMin));
-            end = TimeHelper.convertLocalDateTimeToDate(
-                    TimeHelper.parseLocalDateTime(endDate, endHour, endMin));
-
-            if (start.after(end)) {
+            zone = ZoneId.of(timeZone);
+            start = TimeHelper.parseLocalDateTime(startDate, startHour, startMin);
+            end = TimeHelper.parseLocalDateTime(endDate, endHour, endMin);
+
+            if (start.isAfter(end)) {
```

Comment:
How about changing this to `if (this.rangeStart.isBefore(this.RangeEnd))` and return null early here. Then in the appropriate error block, we just add the specific error messages. Then the same code that appear in every error blocks like `isError = true`, `prepareDefaultPageData` etc. can be unified at the end.